sympy/functions/elementary/hyperbolic.py

```diff
@@ -472,6 +425,46 @@ def _eval_is_real(self):
         re, im = arg.as_real_imag()
         return (im%pi).is_zero
 
+    def _eval_is_positive(self):
+        # cosh(x+I*y) = cos(y)*cosh(x) + I*sin(y)*sinh(x)
```

Comment:
This version is essentially the same but uses generators for laziness. In this version the elementary predicates are only calculated when needed. It works out slightly longer in code but should be more efficient and equally correct.